### PR TITLE
[MAIN] EOS-14243: Adding fru alerts to support bundle

### PR DESCRIPTION
**Pull request by pawankumarsrivastava**
# Backend

## Problem Statement
<pre>
Story Ref (if any): https://jts.seagate.com/browse/EOS-14243
Add FRU Alerts as part of support bundle
</pre>
## Unit testing on RPM done
<pre>
Yes
</pre>
## Problem Description
<pre>
Requirement was to add fru alerts to support bundle.
</pre>
## Solution
<pre>
Added fru alerts to support bundle
</pre>
## Unit Test Cases
<pre>
[root@ssc-vm-c-1058 722996]# cortxcli support_bundle generate -h
usage: cortxcli support_bundle generate [-h] [--os]
                                        [-c {all,csm,sspl,s3server,core,hare,provisioner,os,health_map,manifest,alerts} [{all,csm,sspl,s3server,core,hare,provisioner,os,health_map,manifest,alerts} ...]]
                                        comment

positional arguments:
  comment               Specify the Reason for Generating Support Bundle.

optional arguments:
  -h, --help            show this help message and exit
  --os                  Generate Only OS Logs
  -c {all,csm,sspl,s3server,core,hare,provisioner,os,health_map,manifest,alerts} [{all,csm,sspl,s3server,core,hare,provisioner,os,health_map,manifest,alerts} ...]
                        Generate Bundle for Specific components defined.

[root@ssc-vm-c-1058 722996]# cortxcli support_bundle generate test_alerts6 -c alerts
Please use the below bundle id for checking the status of support bundle.
--------------
| SBtljaort6 |
--------------
Please Find the file on -> /tmp/support_bundle/ .

[root@ssc-vm-c-1058 722996]# cortxcli support_bundle status SBtljaort6
+------------+--------------+--------------------------------+-------------------------------------------------+
| Bundle Id  |   Comment    |           Node Name            |                      Message                    |
+------------+--------------+--------------------------------+-------------------------------------------------+
| SBtljaort6 | test_alerts6 | ssc-vm-c-1059.colo.seagate.com | Tar file linked at location - /tmp/support_bundl|
| SBtljaort6 | test_alerts6 | ssc-vm-c-1059.colo.seagate.com |        Support bundle generation completed.     |
| SBtljaort6 | test_alerts6 | ssc-vm-c-1058.colo.seagate.com |        Support bundle generation completed.     |
| SBtljaort6 | test_alerts6 | ssc-vm-c-1059.colo.seagate.com |        Support bundle generation completed.     |
| SBtljaort6 | test_alerts6 | ssc-vm-c-1058.colo.seagate.com | Tar file linked at location - /tmp/support_bundl|
| SBtljaort6 | test_alerts6 | ssc-vm-c-1058.colo.seagate.com |        Support bundle generation completed.     |
+------------+--------------+--------------------------------+-------------------------------------------------+

[root@ssc-vm-c-1058 722996]# tar -tvf /tmp/support_bundle/SUPPORT_BUNDLE.SBtljaort6
drwxr-xr-x root/root         0 2020-10-18 19:00 SBtljaort6/
drwxr-xr-x root/root         0 2020-10-18 19:00 SBtljaort6/alerts/
-rw-r--r-- root/root      2318 2020-10-18 19:00 SBtljaort6/alerts_SBtljaort6.tar.gz
-rw-r--r-- root/root       137 2020-10-18 19:00 SBtljaort6/summary.yaml
</pre>
Signed-off-by: pawan.kumarsrivastava@seagate.com

### CODE REVIEW

File: cicd/build.sh [hunk 1]
```diff
@@ -215,6 +215,7 @@ cp "$BASE_DIR/cicd/csm_agent.spec" "$TMPDIR"
     CORE_BUILD_START_TIME=$(date +%s)
     mkdir -p "$DIST/csm/conf/service"
     cp "$CONF/setup.yaml" "$DIST/csm/conf"
+    cp "$CONF/alerts_setup.yaml" "$DIST/csm/conf"
```
Comment (navalkp) [inline cicd/build.sh:218]: This will move to section where CLI is build

File: csm/conf/alerts_setup.yaml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+support_bundle:
+    - cortxcli csm_bundle_generate alerts
```
Comment (navalkp) [inline csm/conf/alerts_setup.yaml:2]: This can be moved to csm/cli/conf folder

File: csm/cli/support_bundle/csm_bundle_generate.py [hunk 3]
```diff
@@ -36,14 +41,41 @@ async def init(command):
         :return:
         """
         # Read Config to Fetch Log File Path
-        log_directory_path = Conf.get(const.CSM_GLOBAL_INDEX, "Log.log_path")
+        csm_log_directory_path = Conf.get(const.CSM_GLOBAL_INDEX, "Log.log_path")
+        alerts_file_path = Conf.get(const.CSM_GLOBAL_INDEX, "SUPPORT_BUNDLE.alerts_file_path")
+        # Fetch alerts for support bundle.
```
Comment (prathameshrodi) [inline csm/cli/support_bundle/csm_bundle_generate.py:46]: Fetching Alerts should be done only when component selected will be alert

**Review by prathameshrodi — CHANGES_REQUESTED** [review]

File: csm/cli/support_bundle/csm_bundle_generate.py [hunk 4]
```diff
@@ -36,14 +41,41 @@ async def init(command):
...
+        :return: None
+        """
+        alerts =[]
+        try:
+            conf = GeneralConfig(Yaml(const.DATABASE_CONF).load())
+            db = DataBaseProvider(conf)
+            repo = AlertRepository(db)
+            alerts = await repo.fetch_alert_for_support_bundle()
+        except Exception as ex:
+            Log.exception(f"Error occured while fetching alerts: {ex}")
```
Comment (ajaysrivas) [inline csm/cli/support_bundle/csm_bundle_generate.py:80]: In case of exception, log the error in alerts.json - Internal error : could not fetch alerts.

**Review by ajaysrivas — CHANGES_REQUESTED** [review]

File: cicd/build.sh (hunk 1 above)
Comment (pawankumarsrivastava) [reply cicd/build.sh]: Done

File: csm/conf/alerts_setup.yaml (hunk 2 above)
Comment (pawankumarsrivastava) [reply csm/conf/alerts_setup.yaml]: Done

File: csm/cli/support_bundle/csm_bundle_generate.py (hunk 3 above)
Comment (pawankumarsrivastava) [reply csm/cli/support_bundle/csm_bundle_generate.py]: Done

File: csm/cli/support_bundle/csm_bundle_generate.py (hunk 4 above)
Comment (pawankumarsrivastava) [reply csm/cli/support_bundle/csm_bundle_generate.py:93]: Done

**Review by navalkp — APPROVED** [review]

**Review by ajaysrivas — APPROVED** [review]

**Comment by pawankumarsrivastava** [comment]
Due to separate CLI rpm getting exception for import. Checking now will update the PR.

**Review by prathameshrodi — APPROVED** [review]

**Pull request merged**